plugin/keyring_vault/vault_keys_container.cc

```diff
@@ -0,0 +1,34 @@
+#include <my_global.h>
+#include "vault_keys_container.h"
+
+namespace keyring
+{
+  my_bool Vault_keys_container::init(IKeyring_io* keyring_io, std::string keyring_storage_url)
+  {
+    vault_io = dynamic_cast<IVault_io*>(keyring_io);
```

Comment:
Looks like a hack. Should we extend the interface of `IKeyring_io` with required methods instead?